core/internal/api/api.go

```diff
@@ -63,13 +63,17 @@ type Client interface {
 	Do(*http.Request) (*http.Response, error)
 }
 
+type RetryableClient interface {
```

Comment:
you dont need that httpretryable is alread a Doer interfacw

Reply:
Do you mean the Client right above it? It accepts the wrong type, http.Request vs retryablehttp.Request, and they aren't compatible. I don't see any other `Do` interfaces.

Reply:
hmmm... looks like i was wrong?

Reply:
but why do you need this interface anyways?

Reply:
It's the only way I was able to get a clientImpl that would return `nil, nil`. I spend embarrassingly long, probably at least 40 minutes, trying to write that test in any other way than adding an interface, and I couldn't get anything to work.

Reply:
oh got you. btw were you able to understand what causes the response to be nil?

Reply:
i think the test is kinda trivial so i don't know if there is much value in it, so up to you...

Reply:
No, I never figured it out. Very frustrating. 

It is pretty trivial. I just I didn't like codecov yelling at me for 33% code coverage 😂 I suppose the interface might be useful for other testing (it would have been nice earlier in some other tests, but those had good alternatives), so given it and the code coverage, I'm inclined to keep it. But maybe that's just sunk cost because I spent so long fighting to make it work.